libcontainer/intelrdt/intelrdt.go

```diff
@@ -276,35 +276,30 @@ func findIntelRdtMountpointDir(f io.Reader) (string, error) {
 
 // For Root() use only.
 var (
-	intelRdtRoot string
-	rootMu       sync.Mutex
+	intelRdtRoot    string
+	intelRdtRootErr error
+	rootOnce        sync.Once
 )
 
 // Root returns the Intel RDT "resource control" filesystem mount point.
 func Root() (string, error) {
-	rootMu.Lock()
-	defer rootMu.Unlock()
-
-	if intelRdtRoot != "" {
-		return intelRdtRoot, nil
-	}
-
-	f, err := os.Open("/proc/self/mountinfo")
-	if err != nil {
-		return "", err
-	}
-	root, err := findIntelRdtMountpointDir(f)
-	f.Close()
-	if err != nil {
-		return "", err
-	}
+	rootOnce.Do(func() {
+		f, err := os.Open("/proc/self/mountinfo")
+		if err != nil {
+			intelRdtRootErr = err
+			return
+		}
+		root, err := findIntelRdtMountpointDir(f)
+		f.Close()
+		if err != nil {
+			intelRdtRootErr = err
+			return
+		}
```

Comment:
Do we have to take into account that `resctrl` might get mounted after the first check?

Reply:
I don't think so. If we do, we have to re-read /proc/self/mountinfo over and over (assuming it's some long-lived daemon that imports libcontainer/intelrdt, such as kubelet; for runc itself it's not a issue, since it's a short-lived binary).

I suppose that resctrl is mounted during system init, and it should be mounted by the time we start runc or e.g. kubernetes. If not, it's a configuration issue.

Reply:
K, fair enough